src/tmpfiles/tmpfiles.c

```diff
@@ -2912,16 +3087,37 @@ static int parse_line(
 
         if (!empty_or_dash(age)) {
                 const char *a = age;
+                _cleanup_free_ char *seconds = NULL, *age_by = NULL;
 
                 if (*a == '~') {
                         i.keep_first_level = true;
                         a++;
                 }
 
-                r = parse_sec(a, &i.age);
+                /* Format: "age-by:age"; where age-by is "[abcmABM]+". */
+                if (strchr(a, ':')) {
```

Comment:
so, i don't like if you check first, and then use split_pair() in a second step. you might as well use split_pair() right away, it will tell you via EINVAL that the separator wasn't there.

Reply:
Yup, makes sense. No need for extra check.